mllib/src/main/scala/org/apache/spark/ml/param/sharedParams.scala

```diff
@@ -23,6 +23,27 @@ private[ml] trait HasRegParam extends Params {
   def getRegParam: Double = get(regParam)
 }
 
+private[ml] trait HasElasticNetParam extends HasRegParam {
+  /** param for elastic net regularization parameter */
+  val alphaParam: DoubleParam = new DoubleParam(
```

Comment:
I'd vote for not using HasRegParam at all and using elastic net-specific terminology such as l1RegParam, l2RegParam or something else which makes it clear which param applies to which part of the regularization.